src/Core/Addon/AddonInfo.php

```diff
@@ -14,32 +14,102 @@
  */
 final class AddonInfo
 {
+	/**
+	 * Parse addon comment in search of addon infos.
+	 *
+	 * like
+	 * \code
+	 *   * Name: addon
+	 *   * Description: An addon which plugs in
+	 * . * Version: 1.2.3
+	 *   * Author: John <profile url>
+	 *   * Author: Jane <email>
+	 *   * Maintainer: Jess without link
+	 *   * Maintainer: Robin <email>
+	 *   * Status: in development
+	 * \endcode
+	 *
+	 * @internal Never create this object by yourself, use `Friendica\Core\Addon\AddonHelper::getAddonInfo()` instead.
+	 * @see Friendica\Core\Addon\AddonHelper::getAddonInfo()
+	 *
+	 * @param string $addonId the name of the addon
+	 * @param string $raw The raw file content
+	 */
+	public static function fromString(string $addonId, string $raw): self
+	{
+		$data = [
+			'id' => $addonId,
+		];
+
+		$result = preg_match("|/\*.*\*/|msU", $raw, $m);
+
+		if ($result === false || $result === 0) {
```

Comment:
```suggestion
		if (empty($result) {
```
Seems more appropriate, doesn't it?

Reply:
1. `preg_match()` returns `false` on error and `0` (int) as "no result". So both cases are different, but we are making it explicit that we are aware of both cases and could even handle them differently.
2. `empty()` checks for `""` (empty string), `"0"` (string zero), `0` (integer zero), `0.0` (float zero), `NULL`, `FALSE`, `[]` (empty array) and variables that are not set at all. In our case we know that `$result` could only be `false` or an integer. So checking for `false` and `0` works slightly faster.
3. For strict type safeness it is best practice to compare the types, because it is more robust and precise than using a loose check. This will avoid bugs. Thanks to static code analysis there aren't much places left where I would recommend the use of `empty()` over a strict type comparison.

Reply:
I see. For me, checking something strictly type-safe in PHP feels like possible introduce type-errors in case something isn't properly handled beforehand. Like altering the code before the check and introduce other types.

You say that this will get caught by the static code analysis? Are you fully sure? I found a lot of bugs because of such changes and it feels like we had a lot of bugsfixings because of type-errors (like non-nullable checks for null) before

Reply:
> For me, checking something strictly type-safe in PHP feels like possible introduce type-errors in case something isn't properly handled beforehand.
> [...]
>  I found a lot of bugs because of such changes and it feels like we had a lot of bugsfixings because of type-errors (like non-nullable checks for null) before

That's exactly the reason why strict type checking leads to more robust, predictable code. Using `declare(strict_types=1);)` throws a `TypeError` if a value of the wrong type is passed, halts execution and making the error immediately visible. Type mismatches are caught as soon as they occur, rather than causing unpredictable behavior at runtime. Using loose types will hide bugs and produce unexpected results, especially in larger or more complex codebases.

And yes, type safeness must be introduced slowly while refactoring or introducing new features. That's the reason, why I declare strict types mostly in new files.

> You say that this will get caught by the static code analysis? Are you fully sure?

No, so I checked it. You are right, PHPStan only checks for loose type checking in [strict mode](https://github.com/phpstan/phpstan-strict-rules), but that's not activated right now. Otherwise it would warn about the very loose comparison because of the [empty.notAllowed](https://phpstan.org/error-identifiers/empty.notAllowed) rule, see https://phpstan.org/r/54ed683e-46f4-4ec4-9c4d-a93bd5988c16.

In our case this would only become be a problem if PHP somehow breaks the signature for `preg_match(): int|false`.

Maybe we should care more about the type of `$m`, because it could be `array|null` (PHPStan level 7 [will start complaining](https://phpstan.org/r/c3bcdccf-a21b-4de8-8b73-e117355939f3) about that.)

Reply:
I see and it's ok for me.

Can we activate the strict mode for files with strict_types declaration?

And yes, we should check `$m` as well ;-)

Reply:
> Can we activate the strict mode for files with strict_types declaration?

Strict mode for specific files will require a new phpstan config file. I do not recommend this. I will check in a separate PR if we can activate strict mode in all files and what the consequences would be.

> And yes, we should check `$m` as well ;-)

Done.